src/internal/m365/collection/site/collection.go

```diff
@@ -53,8 +55,9 @@ type prefetchCollection struct {
 	stream map[path.CategoryType]chan data.Item
 	// fullPath indicates the hierarchy within the collection
 	fullPath path.Path
-	// jobs contain the SharePoint.List.IDs or SharePoint.Page.IDs
-	items []string
+	// items contains the SharePoint.List.IDs or SharePoint.Page.IDs
+	// and their corresponding last modified time
+	items map[string]time.Time
```

Comment:
QQ: If both lists and pages are selected to be backed up together, `pc.items` will end up storing IDs for both lists and pages at some point. So if lists follows pages, then we might end up streaming page IDs as list IDs. Could you please confirm?

If true, a quick fix would be to clean up `items` map on `Items()` exit. Other options could be a) 2D map - `path.CategoryType -> map[string]time.Time`. Something similar to the change you made for `stream`. b) use generics like groups does to support channels & group conversations with common code for pages and lists. c) split out pages and lists into separate services.

Reply:
Sorry, my bad. I just looked at the code, this is not possible since `prefetchCollection` instances are different for pages and lists. I was wrong. This is not needed.